platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_general.adoc

```diff
@@ -175,3 +175,205 @@ include::sample[dir="snippets/bestPractices/avoidInternal/groovy",files="build.g
 === Tags
 
 <<tags_reference.adoc#tag:upgrades,`#upgrades`>>
+
+[[modularize_builds]]
+== Modularize Your Builds
+:keywords: structuring-builds
+
+Modularize your builds by creating subprojects.
+
+=== Explanation
+
+Splitting your build's source into multiple modules (Gradle projects) is essential for leveraging Gradle's automatic work avoidance and parallelization features.
```

Comment:
I feel like we should stick to either project or module through this suggestion. We seem to use both here.